apps/web/src/components/Profile/index.tsx

```diff
@@ -49,10 +88,47 @@ const ViewProfile: NextPage = () => {
     return <Custom404 />;
   }
 
-  const profile = data?.profile;
-
   return (
     <>
+      <Modal show={showFollowModal} onClose={() => setShowFollowModal(false)}>
+        <div className="p-5">
+          <div className="flex justify-between text-lg font-bold">
+            <span className="flex">
+              <img
+                onError={({ currentTarget }) => {
+                  currentTarget.src = getAvatar(profile, false);
+                }}
+                src={getAvatar(profile)}
+                className="mr-2 h-10 w-10 rounded-full border bg-gray-200 dark:border-gray-700"
+                alt={formatHandle(profile?.handle)}
+              />
+              <Slug className="flex items-center" slug={formatHandle(profile?.handle)} prefix="@" />{' '}
+            </span>
+            <span className="flex">
+              {followType === 'FeeFollowModuleSettings' ? (
+                <div className="flex space-x-2">
+                  <SuperFollow profile={profile as any} setFollowing={setFollowing} showText />
+                </div>
+              ) : (
+                <div className="flex space-x-2">
+                  <Follow profile={profile as any} setFollowing={setFollowing} showText outline={false} />
+                </div>
+              )}
+              <Button
+                className="ml-3 !px-3 !py-1.5 text-sm"
+                outline
+                onClick={() => {
+                  setShowFollowModal(false);
+                }}
+                aria-label={t`Not now`}
+              >
+                <Trans>Not now</Trans>
+              </Button>
+            </span>
+          </div>
+        </div>
```

Comment:
can we make this as a separate component?

Reply:
Sure, now I moved the stuff that was inside <Modal> to a new component: apps/web/src/components/Profile/FollowDialog.tsx
You can have a look.